src/components/NumberInput/NumberInput.jsx

```diff
@@ -1,57 +1,69 @@
 import { isNumber, toNumber } from 'lodash';
 import React from 'react';
 import PropTypes from 'prop-types';
+import cx from 'classnames';
 import style from './style.scss';
 import {
   getDataAttrs,
   getDataProps,
 } from '../../dataUtils';
 
 class NumberInput extends React.Component {
-
-  constructor() {
-    super();
-    this.onPressIncrement = this.onPressIncrement.bind(this);
-    this.onPressDecrement = this.onPressDecrement.bind(this);
-    this.onChange = this.onChange.bind(this);
-  }
-
-  onPressIncrement() {
-    if (!isNumber(this.props.maxValue) ||
-        this.props.maxValue >= this.props.value + this.props.step) {
-      this.props.onPressIncrement({ diff: this.props.step });
+  onPressIncrement = () => {
+    const { maxValue, value, step, onBlur, onChange } = this.props;
+    if (maxValue >= value + step) {
+      onChange(value + step);
+      onBlur && onBlur();
     }
   }
 
-  onPressDecrement() {
-    if (!isNumber(this.props.minValue) ||
-        this.props.minValue <= this.props.value - this.props.step) {
-      this.props.onPressDecrement({ diff: -1 * this.props.step });
+  onPressDecrement = () => {
+    const { minValue, value, step, onBlur, onChange } = this.props;
+    if (minValue <= value - step) {
+      onChange(value - step);
+      onBlur && onBlur();
     }
   }
 
-  onChange(event) {
-    const value = toNumber(event.nativeEvent.target.value);
-    if (
-      (!this.props.maxValue || value <= this.props.maxValue) &&
-      (!this.props.minValue || value >= this.props.minValue)
-    ) {
-      this.props.onChange({ value });
-    }
+  onChange = (event) => {
+    const { maxValue, minValue, onChange } = this.props;
+    const eventValue = event.nativeEvent.target.value;
+
+    // if the value is cleared, do not convert empty string to 0
+    const value = eventValue === "" ? null : toNumber(eventValue);
+    if (value <= maxValue && value >= minValue) onChange(value);
   }
 
   render() {
+    const { disabled, error, data, placeholder, value, onBlur, onFocus } = this.props;
+
+    const wrapperStyle = cx(style.wrapper, {
+      [style.wrapperDisabled]: disabled,
+      [style.wrapperError]: error,
+    })
+
+    const inputStyle = cx(style.input, {
+      [style.disabled]: disabled,
+    })
+
     return (
       <div
-        {...getDataAttrs(this.props.data)}
-        className={style.wrapper}
+        {...getDataAttrs(data)}
+        className={wrapperStyle}
       >
         <input
           type="number"
-          placeholder={this.props.placeholder}
-          className={style.input}
-          value={this.props.value}
+          placeholder={placeholder}
+          className={inputStyle}
+          disabled={disabled}
+          value={value}
           onChange={this.onChange}
+          onBlur={(e) => {
+            onBlur && onBlur();
```

Comment:
we should not pass anonymous functions if possible.
i think you can just pass onBlur={this.props.onBlur}

Reply:
ah yea..this will be needed if we want the custom styles but can remove for now